ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/util/StreamUtil.java

```diff
@@ -0,0 +1,18 @@
+package uk.gov.hmcts.cmc.ccd.util;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class StreamUtil {
+
+    private StreamUtil() {
+        // Utility class, no instances
+    }
+
+    public static <T> Stream<T> asStream(final Collection<T> collection) {
+        return Optional.ofNullable(collection)
```

Comment:
Yes, we do need this as in CCD domains we can not use Optional. It avoids a null check in `from` method.

Reply:
Unless it’s in lots of places we should probably have the null check, or find a way to make sure we have empty lists

Reply:
at the moment it is more than four places, which motivated me to add this helper method

Reply:
Is us of `Optional.ofNullable(collection)` an extra precaution or we really had problems with null collections?

Reply:
when calling `.stream` on null collection we get `NullPointerException`

Reply:
can't this be solved nicer with a null check around the code?
i.e. this:
```
        builder.bankAccounts(
            asStream(statementOfMeans.getBankAccounts())
                .map(bankAccountMapper::to)
                .filter(Objects::nonNull)
                .map(bankAccount -> CCDCollectionElement.<CCDBankAccount>builder().value(bankAccount).build())
                .collect(Collectors.toList()));
```

could be:
```
if (statementOfMeans.getBankAccounts() != null) {
        builder.bankAccounts(
           statementOfMeans.getBankAccounts()
               .stream()
                .map(bankAccountMapper::to)
                .map(bankAccount -> CCDCollectionElement.<CCDBankAccount>builder().value(bankAccount).build())
                .collect(Collectors.toList()));
}
```

or we initialise it to an empty list rather than using null in the first place

Reply:
above idea is not bad

    public static <T> Stream<T> toStream(final Optional<T> opt) {
        return opt.map(Stream::of).orElse(Stream.empty());
    }

    public static <T> Stream<T> nullableToStream(@Nullable final T object) {
        return toStream(Optional.ofNullable(object));
    }